### ISSUE TITLE
Group OData filter arguments into an object to pass them into the SetHandler actions

### ISSUE THREAD

**Issue by calebkiage** [microsoftgraph/msgraph-cli#32]
Since the `ICommand.SetHandler` function accepts an action with up to 16 arguments, it would help to reduce arguments in our generated code if we created a binding for the already existing `GetQueryParameters` type and excluded query parameters when constructing the handler actions.

The goal is to convert a handler like this:
```csharp
command.SetHandler(async (int? top, int? skip, string search, string filter, bool? count, string[] orderby, string[] select, string[] expand, FormatterType output, IOutputFormatterFactory outputFormatterFactory, CancellationToken cancellationToken) => {
    ...
}, topOption, skipOption, searchOption, filterOption, countOption, orderbyOption, selectOption, expandOption, outputOption);
```

into:
```csharp
command.SetHandler(async (GetQueryParameters getQueryParameters, FormatterType output, IOutputFormatterFactory outputFormatterFactory, CancellationToken cancellationToken) => {
    ...
}, new GetQueryParametersBinder(), outputOption);
```

**Comment by baywet** [comment]
this option will "buy us some time" but not fundamentally solve the issue, if you start to have paths on the API with a lot of parameters it'll still push us over the 16 limitations, even if you saved 8 on the query parameters.

The other aspect to that solution is the experience for the CLI users, they won't be able to specify those things individually anymore as simple strings, they'll have to build some kind of object, which brings another issue, how do we deserialize that object?

**Comment by baywet** [comment]
after [finding the doc](https://github.com/dotnet/command-line-api/blob/main/docs/model-binding.md#more-complex-types) it appears this doesn't rely on any kind of serialization and doesn't change the experience for the end user. Still it wouldn't solve the issue for good, but buy us some time, how much time depends on the API owners themselves and how deep to they nest their APIs.
Their has been a general consensus at the API council level that the more you nest an API, the harder to discover it becomes so API owners generally try to limit nesting.

**Comment by calebkiage** [comment]
So I've found a way to have a collection of types passed into the handler action that doesn't require implementing an additional `ICommandHandler` type. I'm doing this using a binding that accepts a list of `IValueDescriptor` and resolves the actual values in the `GetBoundValue` call

The signature of the generated command handlers changes to:
```csharp
rootCommand.SetHandler<object[]>(parameters => {
    var param0 = parameters[0] as string;
    var param1 = parameters[1] as string;

   ...
}, new CollectionBinding(option1, new LiteralBinding("test"), new TypeBinding(typeof(CancellationToken))));
```

**Comment by calebkiage** [comment]
Fixed in #38
